newIDE/app/src/ExportAndShare/ShareDialog/InviteHome.js

```diff
@@ -0,0 +1,529 @@
+              connection or try again later.
+            </Trans>
+          </PlaceholderError>
+        ) : fetchError === 'project-not-found' ? (
+          <AlertMessage kind="info">
+            <Trans>
+              You need to first save your project to the cloud to invite
+              collaborators.
+            </Trans>
+          </AlertMessage>
+        ) : fetchError === 'project-not-owned' ? (
+          <AlertMessage kind="info">
+            <Trans>
+              You are not owner of this project, so you cannot invite
+              collaborators.
+            </Trans>
+          </AlertMessage>
+        ) : !projectUserAcls ? (
+          <PlaceholderLoader />
+        ) : (
+          projectUserAcls.map(projectUserAcl => (
+            <UserLine
+              username={getCollaboratorUsername(projectUserAcl.userId)}
+              email={projectUserAcl.email}
+              level={projectUserAcl.level}
+              onDelete={() => {
+                doRemoveCollaborator(projectUserAcl);
+              }}
+              disabled={actionInProgress}
+              key={projectUserAcl.userId}
+            />
+          ))
+        )}
+        <Line>
+          <Column expand noMargin>
+            <RaisedButton
+              icon={<Add fontSize="small" />}
+              label={<Trans>Add collaborator</Trans>}
+              onClick={() => setShowCollaboratorAddDialog(true)}
+              primary
+              disabled={
+                !hasSufficientPermissionsWithSubscription ||
+                !!fetchError ||
+                !projectUserAcls ||
+                actionInProgress
+              }
+            />
+          </Column>
+        </Line>
+      </ColumnStackLayout>
+      {showCollaboratorAddDialog && (
+        <Dialog
+          title="Add a collaborator"
+          actions={[
+            <FlatButton
+              label={<Trans>Back</Trans>}
+              disabled={actionInProgress}
+              key="back"
+              primary={false}
+              onClick={() => {
+                if (!isEmailValid) setIsEmailValid(true);
+                if (addError) setAddError(null);
+                setShowCollaboratorAddDialog(false);
+              }}
+            />,
+            <LeftLoader isLoading={actionInProgress} key="add-collaborator">
+              <DialogPrimaryButton
+                label={<Trans>Add</Trans>}
+                primary
+                onClick={doAddCollaborator}
+                disabled={actionInProgress}
+              />
+            </LeftLoader>,
+          ]}
+          maxWidth="xs"
+          cannotBeDismissed={actionInProgress}
+          onRequestClose={() => setShowCollaboratorAddDialog(false)}
+          onApply={doAddCollaborator}
+          open
+        >
+          <form
+            onSubmit={event => {
+              // Prevent browser to navigate on form submission.
+              event.preventDefault();
```

Comment:
Should we at some point make a `UI/Form`  to encapsulate this behavior?

Reply:
Yeah I'll do it now.

Reply:
Form component created and used